api/src/test/java/org/pmiops/workbench/cdr/CdrVersionServiceTest.java

```diff
@@ -139,98 +145,109 @@ public void setUp() {
 
   @Test
   public void testSetCdrVersionDefault() {
-    when(fireCloudService.isUserMemberOfGroup(
-            user.getUsername(), registeredTier.getAuthDomainName()))
-        .thenReturn(true);
+    addMembershipForTest(registeredTier);
     cdrVersionService.setCdrVersion(defaultCdrVersion);
     assertThat(CdrVersionContext.getCdrVersion()).isEqualTo(defaultCdrVersion);
   }
 
   @Test
   public void testSetCdrVersionDefaultId() {
+    addMembershipForTest(registeredTier);
+    cdrVersionService.setCdrVersion(defaultCdrVersion.getCdrVersionId());
+    assertThat(CdrVersionContext.getCdrVersion()).isEqualTo(defaultCdrVersion);
+  }
+
+  @Test(expected = ForbiddenException.class)
+  public void testSetCdrVersionDefaultForbiddenNotInTier() {
+    cdrVersionService.setCdrVersion(defaultCdrVersion);
+  }
+
+  @Test(expected = ForbiddenException.class)
+  public void testSetCdrVersionDefaultIdForbiddenNotInTier() {
+    cdrVersionService.setCdrVersion(defaultCdrVersion.getCdrVersionId());
+  }
+
+  // these tests fail because the user is in the right tier according to the AoU DB
+  // but the user is not in the right auth domain according to Terra
+
+  @Test(expected = ForbiddenException.class)
+  public void testSetCdrVersionDefaultForbiddenNotInGroup() {
+    accessTierService.addUserToTier(user, registeredTier);
+
     when(fireCloudService.isUserMemberOfGroup(
             user.getUsername(), registeredTier.getAuthDomainName()))
-        .thenReturn(true);
+        .thenReturn(false);
+
+    cdrVersionService.setCdrVersion(defaultCdrVersion);
+  }
+
+  @Test(expected = ForbiddenException.class)
+  public void testSetCdrVersionDefaultIdForbiddenNotInGroup() {
+    accessTierService.addUserToTier(user, registeredTier);
+
+    when(fireCloudService.isUserMemberOfGroup(
+            user.getUsername(), registeredTier.getAuthDomainName()))
+        .thenReturn(false);
+
     cdrVersionService.setCdrVersion(defaultCdrVersion.getCdrVersionId());
-    assertThat(CdrVersionContext.getCdrVersion()).isEqualTo(defaultCdrVersion);
   }
 
   @Test
   public void testSetCdrVersionControlled() {
-    when(fireCloudService.isUserMemberOfGroup(
-            user.getUsername(), controlledTier.getAuthDomainName()))
-        .thenReturn(true);
-    config.featureFlags.unsafeAllowAccessToAllTiersForRegisteredUsers = true;
```

Comment:
This flag no longer affects CdrVersionService in any way; the service follows what the AccessTierService says about the user.